src/metabase/native_query_analyzer.clj

```diff
@@ -49,11 +49,32 @@
    ;; (t2/table-name :model/Table) doesn't work on CI since models/table.clj hasn't been loaded
    :join [[:metabase_table :t] [:= :table_id :t.id]]})
 
+;; NOTE: be careful when adding square braces, as the rules for nesting them are different.
+(def ^:private quotes "\"`")
+
+(defn- quote-stripper
+  "Construct a function which unquotes values which use the given character as their quote."
+  [quote-char]
+  (let [doubled (str quote-char quote-char)
+        single  (str quote-char)]
+    #(-> (subs % 1 (dec (count %)))
+         (str/replace doubled single))))
+
+(def ^:private quote-strippers
+  "Pre-constructed lambdas, to save some memory allocations."
+  (zipmap quotes (map quote-stripper quotes)))
+
 (defn- field-query
   "Exact match for quoted fields, case-insensitive match for non-quoted fields"
   [field value]
-  (if (= (first value) \")
-    [:= field (-> value (subs 1 (dec (count value))) (str/replace "\"\"" "\""))]
+  (if-let [quote-stripper (quote-strippers (first value))]
+    [:= field (quote-stripper value)]
+    ;; Technically speaking this is not correct for all databases.
+    ;; For example Oracle treats non-quoted identifiers as uppercase, but still expects a case sensitive match.
```

Comment:
I guess it makes sense to clarify that PG is doing the same, but lowercase?

Reply:
I think one exception is enough to break the rule? 

I guess it would be nice to say that this logic *is* correct for MySQL, and then I can mention a few.

Reply:
Reworded to mention a few database twists and turns